omc3/scripts/linfile_clean.py

```diff
@@ -0,0 +1,209 @@
+        ),
+        # restore backup
+        restore=dict(
+            action="store_true",
+            help=("Restore the latest backup of the files. "
+                  "If this parameter is given, no cleaning is performed."),
+        ),
+        # for actual cleaning
+        columns=dict(
+            nargs='+',
+            type=str,
+            help="Columns to clean on.",
+        ),
+        limit=dict(
+            type=float,
+            help="Do not clean data-points deviating less than this limit from the average.",
+            default=0.0,
+        ),
+        backup=dict(
+            action="store_true",
+            help=("Create a backup of the files before overwriting. "
+                  "The backups are numbered, with the highest number being "
+                  "the latest backup.")
+        ),
+    )
+
+
+@entrypoint(get_params(), strict=True)
+def main(opt):
+    """Main function, to parse commandline input and separate restoration
+    from cleaning."""
+    save_config(Path('.'), opt, __file__)
+    if opt.restore:
+        restore_files(opt.files)
+        return
+
+    if opt.columns is None:
+        raise ValueError("The option 'columns' is required for cleaning.")
+    clean_columns(opt.files, opt.columns, opt.limit, opt.backup)
+
+
+# Restore ----------------------------------------------------------------------
+
+def restore_files(files: Sequence[Union[Path, str]]):
+    """Restore backupped files."""
+    failed = []
+    for file in files:
+        file = Path(file)
+        try:
+            _restore_file(file)
+        except IOError as e:
+            failed.append(str(e))
+
+    if len(failed):
+        all_errors = '\n'.join(failed)
+        if len(failed) == len(files):
+            raise IOError(f"Restoration of ALL files has failed\n{all_errors}")
+        raise IOError(f"Restoration of some files has failed, "
+                      f"but the others were restored:\n{all_errors}")
+
+    LOG.info("Restoration successfully completed.")
+
+
+def _restore_file(file):
+    counter = 1
+    backup_file = _get_backup_filepath(file, counter)
+    if not backup_file.exists():
+        raise IOError(f"No backups found for file {file.name}")
+
+    # get last existing backup file
+    while backup_file.exists():
+        counter += 1
+        backup_file = _get_backup_filepath(file, counter)
+    backup_file = _get_backup_filepath(file, counter-1)
+
+    # restore found  file
+    LOG.info(f"Restoring last backup file {backup_file.name} to {file.name}.")
+    shutil.move(backup_file, file)  # overwrites file
+
+
+# Clean ------------------------------------------------------------------------
+
+def clean_columns(files: Sequence[Union[Path, str]], columns: Sequence[str],
+                  limit: float = 0.0, backup: bool = True):
+    """ Clean the columns in the given files."""
+    for file in files:
+        file = Path(file)
+        LOG.info(f"Cleaning {file.name}.")
```

Comment:
Probably `debug` too. Could be a lot of logging output if a bunch of files are given

Reply:
yes. That's on purpose. It's not a function to be used, but a script.

Reply:
Let's think about this though. Given the amount of `info` level statements, if you give 5 files and 3 columns, you're already at 5 * 3 * 2 = 30 `info` level lines. This feels like polluting the output to me and drowning what is relevant information.

I'm ok with this specific line being `LOG.info` as long as the other ones are `debug` for instance.

Reply:
I'm all in favour of keeping the output minimal for functions. But this is a `script` that should be called from commandline and I want to see what has happened in here to my files. If on the other side, I activate debug loggig, THEN it will be super polluted, but all the downstream functions logging things.

Reply:
:man_shrugging: